src/promptflow/tests/sdk_cli_azure_test/recording_utilities/bases.py

```diff
@@ -119,7 +119,18 @@ def _process_response_recording(self, response: Dict) -> Dict:
         if is_live():
             return response
 
-        response["body"]["string"] = response["body"]["string"].decode("utf-8")
+        # sync and async responses have different structure
+        # sync has .body.string, while async has .content
+        # in our sanitizers (processors) logic, we only handle .body.string
+        # so make async align sync for less code change
+        is_async_response = False
+        if "body" not in response:
```

Comment:
do we have a better choice to judge if this is an async response? This seems fragil

Reply:
I think it will be better if judge from key `content`, which seems a contract for httpx, which is safer to use.